src/eth/accounts/setup.js

```diff
@@ -32,28 +32,17 @@ export async function getBrowserProvider() {
     return subprovider;
   };
 
-  // If web3 is injected (old MetaMask)...
-  if (typeof window.web3 !== 'undefined') {
-    return wrap(window.web3.currentProvider);
-  }
-
-  // If web3 is not injected (new MetaMask)...
-  return new Promise((resolve, reject) => {
-    let resolved = false;
-
-    window.addEventListener('message', ({ data }) => {
-      if (data && data.type && data.type === 'ETHEREUM_PROVIDER_SUCCESS') {
-        resolved = true;
+  return new Promise(async (resolve, reject) => {
```

Comment:
Since MetaMask dropped the `postMessage` -> `addEventListener('message', ...)` way of doing things, this wrapper promise is no longer necessary -- instead of `resolve`, just `return` will do